src/index.js

```diff
@@ -1,42 +1,57 @@
 // ITERATION 1
 
+
+
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  let price = product.querySelector(".price span");
+  let quantity = product.querySelector(".quantity input");
+  let subtotal = product.querySelector(".subtotal span");
+  let total = price.innerHTML * quantity.value;
+  return subtotal.innerHTML = total; 
 }
 
-function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
 
-  // ITERATION 2
-  //... your code goes here
+function calculateAll() {
 
-  // ITERATION 3
-  //... your code goes here
+  let products = document.getElementsByClassName('product');
+  let sumOfProducts = 0;
+  for(let i = 0; i < products.length ; i++) {
+    sumOfProducts += updateSubtotal(products[i]);
+  }
+  
+  let finalTotal = document.querySelector("#total-value span");
+  finalTotal.innerHTML = sumOfProducts;
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  console.log('The target in remove is:', target.parentNode.parentNode);
+  target.parentNode.parentNode.remove();
+  calculateAll();
 }
 
 // ITERATION 5
+// couldn´t figure out this one yet :/
 
 function createProduct() {
```

Comment:
This one was a little bit tricky, I will leave here the solution with comments that explain what every line of code is doing:

```
function createProduct() {
  // Targeting all the create product inputs that the user will fill
  let newProductInputs = document.querySelectorAll(".create-product input");

  // Save the value of the inputs in variables
  let newProductName = newProductInputs[0].value;
  let newProductPrice = newProductInputs[1].value;

  // Saving the newProduct html code in a variable
  let newProduct = `
  <tr class="product">
    <td class="name">
      <span>${newProductName}</span>
    </td>
    <td class="price">$<span>${newProductPrice}</span></td>
    <td class="quantity">
      <input type="number" value="0" min="0" placeholder="Quantity" />
    </td>
    <td class="subtotal">$<span>0</span></td>
    <td class="action">
      <button class="btn btn-remove">Remove</button>
    </td>
  </tr>`;

  // Targeting the table
  const tableBody = document.querySelector("tbody");

  // Adding the newProduct to the end of the table
  tableBody.insertAdjacentHTML("beforeend", newProduct);

  // Adding the removeProduct function to the remove button of the new product
  let removeBtn = document.querySelectorAll(".btn-remove");
  removeBtn.forEach((btn) => btn.addEventListener("click", removeProduct));
}
```